crates/ruff_linter/src/rules/flake8_bugbear/rules/batched_without_explicit_strict.rs

```diff
@@ -0,0 +1,101 @@
+use crate::checkers::ast::Checker;
+use crate::fix::edits::add_argument;
+use crate::rules::flake8_bugbear::rules::is_infinite_iterator;
+use crate::settings::types::PythonVersion;
+use ruff_diagnostics::{AlwaysFixableViolation, Applicability, Diagnostic, Fix};
+use ruff_macros::{derive_message_formats, ViolationMetadata};
+use ruff_python_ast::{Arguments, ExprCall};
+
+/// ## What it does
+/// Checks for `itertools.batched` calls without an explicit `strict` parameter.
+///
+/// ## Why is this bad?
+/// By default, if the length of the iterable is not divisible by
+/// the second argument to `itertools.batched`, the last batch
+/// will be shorter than the rest.
+///
+/// Pass `strict=True` to raise a `ValueError` if the batches are of non-uniform length.
+/// Otherwise, pass `strict=False` to make the intention explicit.
+///
+/// ## Example
+/// ```python
+/// itertools.batched(iterable, n)
+/// ```
+///
+/// Use instead:
+/// ```python
+/// itertools.batched(iterable, n, strict=True)
+/// ```
+///
+/// ## Fix safety
+/// This rule's fix is marked as unsafe for `batched` calls that contain
+/// `**kwargs`, as adding a `strict` keyword argument to such a call may lead
+/// to a duplicate keyword argument error.
+///
+/// ## References
+/// - [Python documentation: `batched`](https://docs.python.org/3/library/itertools.html#batched)
+#[derive(ViolationMetadata)]
+pub(crate) struct BatchedWithoutExplicitStrict;
+
+impl AlwaysFixableViolation for BatchedWithoutExplicitStrict {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        "`batched()` without an explicit `strict=` parameter".to_string()
+    }
+
+    fn fix_title(&self) -> String {
+        "Add explicit value for parameter `strict=`".to_string()
+    }
+}
+
+/// B911
+pub(crate) fn batched_without_explicit_strict(checker: &mut Checker, call: &ExprCall) {
+    if checker.settings.target_version < PythonVersion::Py313 {
+        return;
+    }
+
+    let semantic = checker.semantic();
+    let (func, arguments) = (&call.func, &call.arguments);
+
+    let Some(qualified_name) = semantic.resolve_qualified_name(func) else {
+        return;
+    };
+    let first_positional = arguments.args.first();
```

Comment:
It seems we need to be careful around starred expressions

```suggestion
    let first_positional = arguments.find_positional(0);
```